cmd/exec_common/shared.go

```diff
@@ -0,0 +1,54 @@
+package exec_common
+
+import (
+	"fmt"
+	"strings"
+)
+
+// DatastoreEngine type to define different engine types
+type DatastoreEngine string
+
+// Types of Datastore Engines
+const (
+	Memory   DatastoreEngine = "memory"
+	Postgres DatastoreEngine = "postgres"
+	MySQL    DatastoreEngine = "mysql"
+)
+
+func (e DatastoreEngine) String() string {
+	return string(e)
+}
+
+func NewDatastoreEngine(engine string) (*DatastoreEngine, error) {
+	var isValid bool
+	var validEngine DatastoreEngine
+	for _, engineType := range []DatastoreEngine{Memory, Postgres, MySQL} {
+		if engineType.String() == engine {
+			isValid = true
+			validEngine = engineType
+		}
+	}
+	if !isValid {
+		return nil, fmt.Errorf("invalid datastore engine (%s)", engine)
+	}
+	return &validEngine, nil
+}
+
+func VerifyDatastoreEngine(uri string, engine DatastoreEngine) error {
+	var URIType string
+	URITokens := strings.Split(uri, ":")
+	if len(URITokens) > 1 {
+		URIType = URITokens[0]
+		switch URIType {
```

Comment:
These uris are not correctly defined. We have `postgres://...` and `mysql://...`, not `postgresql://` and `mysqlx`

Please also fix the corresponding tests in `run/run_test.go`.

Reply:
`./openfga run --datastore-engine mysql --datastore-uri 'postgres://postgres:password@localhost:5432/postgres'` should return this validation error but currently does not as a result of this.

Also, I'm confused by the `mysqlx://` bit.. If you choose to run OpenFGA with MySQL, then your datastore connection settings wouldn't involve a datastore URI.

`./openfga run --datastore-engine mysql --datastore-uri 'root:secret@tcp(localhost:3306)/openfga'`

Are we intending to change the overall CLI behavior to require a datastore uri scheme as a prefix? It seems that is what would be required to make this functionality work as intended.

Reply:
`mysqlx` was a mistake on my end, which I have amended. It is applicable to `X` plugin, which I dont so is supported. 

Also about your 2nd comment, which was regarding `mysql` engine not requiring `mysql*` type URI is handled and I fixed tests too 

Let me know if you think any more additions are required @jon-whit